ethereum/core/src/test-support/java/tech/pegasys/pantheon/ethereum/core/BlockDataGeneratorWithReason.java

```diff
@@ -0,0 +1,25 @@
+/*
+ * Copyright 2019 ConsenSys AG.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License is distributed on
+ * an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations under the License.
+ */
+package tech.pegasys.pantheon.ethereum.core;
+
+import java.util.Arrays;
+import java.util.Optional;
+
+public class BlockDataGeneratorWithReason extends BlockDataGenerator {
```

Comment:
Could this just be done with Mocking?

Reply:
Could be but theres a BlockDataGenerator that already exists. And the code is nicer using the data generator rather than adding mock. I've also added a receipt(revertReason) method to the existing BlockDataGenerator remove need for this extra BlockDataGenerator class.